packages/amplify-category-hosting/__tests__/index.test.js

```diff
@@ -0,0 +1,214 @@
+jest.mock('promise-sequential');
+
+const inquirer = require('inquirer');
+const mockirer = require('mockirer');
+const sequential = require('promise-sequential');
+
+const categoryManager = require('../lib/category-manager');
+
+const indexModule = require('../index');
+
+describe('index', () => {
+    const mockContext = {}; 
+    const S3AndCloudFront = 'S3AndCloudFront'; 
+    const AnotherHostingService = 'AnotherHostingService';
+
+    const mockAvailableServices = []; 
+    let mockDisabledServices = []; 
+    let mockEnabledServices = []; 
+
+    const mockAnswers = {
+        selectedServices: []
+    }; 
+
+    beforeAll(() => {
+        categoryManager.getCategoryStatus = jest.fn(()=>{
+            return {
+                availableServices: mockAvailableServices,
+                enabledServices: mockEnabledServices,
+                disabledServices: mockDisabledServices
+            }
+        }); 
+        categoryManager.runServiceAction = jest.fn(); 
+        categoryManager.migrate = jest.fn(); 
+    }); 
+
+    beforeEach(() => {
+        mockAvailableServices.length = 0; 
```

Comment:
don't know how that can be done, because the behavior of the methods, depends on the return of the getCategoryStatus method, not mocking its return wouldn't allow us to test all the different scenarios.